internal/formats/spdx22json/to_format_model.go

```diff
@@ -219,19 +240,32 @@ func toFileTypes(metadata *source.FileMetadata) (ty []string) {
 func toRelationships(relationships []artifact.Relationship) (result []model.Relationship) {
 	for _, r := range relationships {
 		exists, relationshipType, comment := lookupRelationship(r.Type)
-
 		if !exists {
 			log.Warnf("unable to convert relationship from SPDX 2.2 JSON, dropping: %+v", r)
 			continue
 		}
 
-		result = append(result, model.Relationship{
-			SpdxElementID:      model.ElementID(r.From.ID()).String(),
-			RelationshipType:   relationshipType,
-			RelatedSpdxElement: model.ElementID(r.To.ID()).String(),
-			Comment:            comment,
-		})
+		rel := model.Relationship{
+			SpdxElementID:    model.ElementID(r.From.ID()).String(),
+			RelationshipType: relationshipType,
+			Comment:          comment,
+		}
+
+		// if this relationship contains an external document ref, we need to use DocElementID instead of ElementID
+		if _, ok := r.To.(rekor.ExternalRef); ok {
+			if r.Type == artifact.DescribedByRelationship {
```

Comment:
nit: small style consideration here - the negative path might be better to express first while removing the secondary if/else block

Since `continue` with the negative case has been used above successfully consider
```
_, ok := r.To.(rekor.ExternalRef)
if !ok {
	rel.RelatedSpdxElement = model.ElementID(r.To.ID()).String()
	result = append(result, rel)
	continue
}

if r.Type == artifact.DescribedByRelationship {
	rel.RelatedSpdxElement = model.DocElementID(r.To.ID()).String()
	result = append(result, rel)
	continue
}

log.Warnf("Syft does not know how to describe the external reference relationship in SPDX JSON, dropping %+v", r)
```

While the append and continue are repeated a bit, we lose the nested if/else structure and make the drop our default case if the type conversion is unsuccessful, OR if `DescribeByRelationship` is not the clear type

Reply:
I agree, that looks better. 

How about this? toRelationships and toExternalDocumentRefs both operate on relationships with ExternalRefs, and I think it would be good to enforce consistency in how they are handled in both functions, by delegating to another function. 

```
// if this relationship contains an external document ref, we need to use DocElementID instead of ElementID
valid, err := isValidExternalRelationshipDocument(r)
if err != nil {
	log.Warnf("dropping relationship %v: %w", rel, err)
	continue
}
if valid {
	rel.RelatedSpdxElement = model.DocElementID(r.To.ID()).String()
} else {
	rel.RelatedSpdxElement = model.ElementID(r.To.ID()).String()
}

result = append(result, rel)
```
where

```
// isValidExternalRelationshipDocument returns if rel contains an ExternalRef and if it to_format_model know how to handle it.
// An error is returned if rel contains an ExternalRef, but the rel cannot be handled
func isValidExternalRelationshipDocument(rel artifact.Relationship) (bool, error) {
	if _, ok := rel.From.(rekor.ExternalRef); ok {
		return false, errors.New("syft cannot handle an ExternalRef in the FROM field of a relationship")
	}
	if _, ok := rel.To.(rekor.ExternalRef); ok {
		relationshipType := artifact.DescribedByRelationship
		if rel.Type == relationshipType {
			return true, nil
		}
		return false, fmt.Errorf("syft cannot handle an ExternalRef with relationship type: %v", relationshipType)
	}
	return false, nil
}
```

and similar change to toExternalDocumentRefs

Reply:
Awesome!